File: examples/functional-equivalence-output/pages/inventory.page.ts
import { Page, Locator, expect } from '@playwright/test';

export class InventoryPage {
  readonly page: Page;
  readonly pageTitle: Locator;
  readonly firstInventoryItem: Locator;
  readonly cartLink: Locator;

  constructor(page: Page) {
    this.page = page;
    this.pageTitle = page.locator(".title");
    // [MANUAL EDIT #2 — Selenium/Playwright strict-mode gap]
    // Java `@FindBy(css=".inventory_item")` implicitly took the first match
    // (Selenium `findElement` semantics). Playwright's `page.locator(...)`
    // matches all 6 inventory items; `.isVisible()` on that ambiguous
    // locator throws "strict mode violation". Adding `.first()` restores
    // the original single-element intent. A future sel2pw patch could
    // detect this idiom automatically when `.isDisplayed()` is called on
    // a multi-match locator; for now, a one-line hand edit.
    this.firstInventoryItem = page.locator(".inventory_item").first();
    this.cartLink = page.locator("#shopping_cart_container");
  }

  async getPageTitle(): Promise<string> {
    return await this.pageTitle.innerText();
  }

  async isInventoryVisible(): Promise<boolean> {
    return await this.firstInventoryItem.isVisible();
  }

  async isCartVisible(): Promise<boolean> {
    return await this.cartLink.isVisible();
  }
}

File: examples/functional-equivalence-target/src/main/java/com/sel2pw/eval/pages/InventoryPage.java
package com.sel2pw.eval.pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.PageFactory;

/**
 * Page object for the Sauce Labs demo inventory (post-login landing) page.
 */
public class InventoryPage {

    private final WebDriver driver;

    @FindBy(css = ".title")
    private WebElement pageTitle;

    @FindBy(css = ".inventory_item")
    private WebElement firstInventoryItem;

    @FindBy(id = "shopping_cart_container")
    private WebElement cartLink;

    public InventoryPage(WebDriver driver) {
        this.driver = driver;
        PageFactory.initElements(driver, this);
    }

    public String getPageTitle() {
        return pageTitle.getText();
    }

    public boolean isInventoryVisible() {
        return firstInventoryItem.isDisplayed();
    }

    public boolean isCartVisible() {
        return cartLink.isDisplayed();
    }
}
